### PR TITLE
jwt-auth: make audiences configurable for each issuer

### PR DESCRIPTION
**Pull request by mtakigiku**

### CODE REVIEW

File: src/envoy/auth/config.cc [hunk 1]
```diff
@@ -158,6 +167,18 @@ IssuerInfo::IssuerInfo(Json::Object *json) {
   failed_ = true;
 }
 
+bool IssuerInfo::IsValidAudience(const std::string &aud) {
+  if (audiences_.empty()) {
+    return true;
+  }
+  for (auto &a : audiences_) {
```
Comment (lizan) [inline src/envoy/auth/config.cc]: variable name `a` is not good, name it `audience`?

File: src/envoy/auth/config.cc [hunk 2]
```diff
@@ -158,6 +167,18 @@ IssuerInfo::IssuerInfo(Json::Object *json) {
   failed_ = true;
 }
 
+bool IssuerInfo::IsValidAudience(const std::string &aud) {
```
Comment (lizan) [inline src/envoy/auth/config.cc]: `IsAudienceAllowed`

Comment (mtakigiku) [reply src/envoy/auth/config.cc]: done

File: src/envoy/auth/config.cc (hunk 1 above)
Comment (mtakigiku) [reply src/envoy/auth/config.cc]: resolved by using std::find()

**Pull request merged**